themis/languages/java/installation-android.md

```diff
@@ -35,6 +35,33 @@ If you experience difficulties with jcenter repository, try using maven instead:
 
 Once JavaThemis is installed, you can [try out examples on your machine](../examples/).
 
+
+## Configuring ProGuard / R8 rules
+
+Sometimes when working with Android projects in `Release` configuration, you might get runtime error, like:
+
+```
+Java.Lang.IncompatibleClassChangeError: no non-static method "Lcom/cossacklabs/themis/SecureCellSeal;.decrypt([B[B)[B"
+[orion.mobile]   at Java.Interop.JniEnvironment+InstanceMethods.GetMethodID (Java.Interop.JniObjectReference type, System.String name, System.String signature) [0x0005b] in <42d2b7086f0a46efb99253c5db1ecca9>:0 
+[orion.mobile]   at Android.Runtime.JNIEnv.GetMethodID (System.IntPtr kls, System.String name, System.String signature) [0x00007] in <3080427739614e60a939a88bf3f838d5>:0 
+[orion.mobile]   at Com.Cossacklabs.Themis.SecureCell+ISealInvoker.Decrypt (System.Byte[] p0, System.Byte[] p1) [0x00017] in <cd618986d1ce4194b63cdd3366dad291>:0 
+[orion.mobile]   at Themis.Droid.CellSealDroid.UnwrapData (Themis.ISecureCellData cipherTextData, System.Byte[] context) [0x0007e] in <a492e7118e094c3296442a386fe5d80e>:0 
+[orion.mobile]    --- End of inner exception stack trace --
+```
+
+It is caused by "over-optimizing" a build by ProGuard / R8, which strips native Java classes from a build. 
```

Comment:
Rephrased a bit so that the "blame" falls on ProGuard, rather then the developers who "over-optimized their build".

```suggestion
It is caused by ProGuard / R8 "over-optimizing" a build, stripping native Java classes as a result.
```